pkg/yang/entry.go

```diff
@@ -1519,17 +1550,40 @@ func errorSort(errors []error) []error {
 	return errors[:i]
 }
 
-// DefaultValue returns the schema default value for e, if any. If the leaf
-// has no explicit default, its type default (if any) will be used.
-func (e *Entry) DefaultValue() string {
-	if e.Default != "" {
-		return e.Default
-	} else if typ := e.Type; typ != nil {
-		if leaf, ok := e.Node.(*Leaf); ok {
-			if leaf.Mandatory == nil || leaf.Mandatory.Name == "false" {
-				return typ.Default
+// HasDefault indicates whether the leaf entry has a default value or not.
+func (e *Entry) HasDefault() bool {
+	return len(e.DefaultValues()) > 0
+}
+
+// SingleDefaultValue returns the schema default value for e, if any. This is
```

Comment:
I wonder whether rather than requiring a user to do two calls here - i.e.,:

```golang

var default string
if a.HasDefault() {
   default = a.SingledefaultValue()
}
```

it's better to have these combined, such that you can do:

```
if def, ok := a.SingleDefaultValue(); a.IsLeaf() && ok {
   default = def
}
```

Since `DefaultValues` will return `nil`, then this is detectable in a single call for the `leaf-list` case so this approach would create some symmetry between the `leaf` and `leaf-list` types.

Reply:
Doing it this way seems better for usability as well since it makes error handling more natural. I'll make the change.

Reply:
Note that `Entry.SingleDefaultValue() (string, bool)` only returns true when there is one and only one default value. This includes the case when a leaf-list has a single default value.

Reply:
SGTM!